### PR TITLE
118: upgrade dependencies for CVE + update PK

### LINKED ISSUE #118: Upgrade dependencies to fix security issue in Postgres JDBC driver
+ Upgrade dependencies to fix security issue in Postgres JDBC driver
+ Update project to PK2

### PR DESCRIPTION
**Pull request by pj-spoelders**
Closes #118

### CODE REVIEW

File: .github/workflows/ci-build.yml [hunk 1]
```diff
@@ -1,55 +1,55 @@
...
+  pull_request:
 
 jobs:
   build:
-    strategy:
-      fail-fast: false
-      matrix:
-        docker_db_version: [ 7.0.16, 7.1.6 ]
-    env:
-      DEFAULT_DB_VERSION: "7.1.6"
```
Comment (kaklakariada) [inline .github/workflows/ci-build.yml]: Restore the matrix build, upgrade to latest versions

File: .github/workflows/ci-build.yml [hunk 2]
```diff
@@ -1,55 +1,55 @@
...
-        if: ${{ env.DEFAULT_DB_VERSION == matrix.docker_db_version }}
-        run: mvn --batch-mode sonar:sonar -Dsonar.organization=exasol -Dsonar.host.url=https://sonarcloud.io -Dsonar.login=$SONAR_TOKEN
+        if: ${{ env.SONAR_TOKEN != null }}
+        run: |
+          mvn --batch-mode org.sonarsource.scanner.maven:sonar-maven-plugin:sonar \
+              -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn \
+              -DtrimStackTrace=false \
+              -Dsonar.organization=exasol \
+              -Dsonar.host.url=https://sonarcloud.io \
+              -Dsonar.login=$SONAR_TOKEN
```
Comment (kaklakariada) [inline .github/workflows/ci-build.yml]:
```suggestion
          mvn --batch-mode org.sonarsource.scanner.maven:sonar-maven-plugin:sonar \
              -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn \
              -DtrimStackTrace=false \
              -Dsonar.organization=exasol \
              -Dsonar.host.url=https://sonarcloud.io \
              -Dsonar.login=$SONAR_TOKEN \
              -Dcom.exasol.dockerdb.image=${{ matrix.docker_db_version }}
```

File: src/assembly/all-dependencies.xml [hunk 3]
```diff
@@ -14,22 +14,9 @@
...
-            </unpackOptions>
             <scope>runtime</scope>
+            <outputDirectory>/</outputDirectory>
         </dependencySet>
     </dependencySets>
-    <fileSets>
-        <fileSet>
-            <directory>${project.build.outputDirectory}</directory>
-            <outputDirectory>/</outputDirectory>
-        </fileSet>
```
Comment (kaklakariada) [inline src/assembly/all-dependencies.xml]: Not sure if we need to restore this and add an exception to project-keeper?

File: pom.xml [hunk 4]
```diff
@@ -3,20 +3,64 @@
...
     <groupId>com.exasol</groupId>
     <artifactId>row-level-security</artifactId>
-    <version>3.0.3</version>
+    <version>3.0.4</version>
     <name>Exasol Row Level Security</name>
     <description>This projects provides a plug-in to the Exasol database that adds per-row access control.</description>
     <properties>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
         <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
         <java.version>11</java.version>
```
Comment (kaklakariada) [inline pom.xml]:
```suggestion
```
Already contained in parent pom

**Review by kaklakariada — CHANGES_REQUESTED** [review]

File: src/assembly/all-dependencies.xml (hunk 3 above)
Comment (pj-spoelders) [reply src/assembly/all-dependencies.xml]: did so

File: .github/workflows/ci-build.yml [hunk 5]
```diff
@@ -48,15 +48,9 @@ jobs:
...
-        run: |
-          mvn --batch-mode org.sonarsource.scanner.maven:sonar-maven-plugin:sonar \
-              -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn \
-              -DtrimStackTrace=false \
-              -Dsonar.organization=exasol \
-              -Dsonar.host.url=https://sonarcloud.io \
-              -Dsonar.login=$SONAR_TOKEN
-              -Dcom.exasol.dockerdb.image=${{ matrix.docker_db_version }}
+      - name: Sonar analysis
+        if: ${{ env.DEFAULT_DB_VERSION == matrix.docker_db_version }}
```
Comment (kaklakariada) [inline .github/workflows/ci-build.yml]:
```suggestion
        if: ${{ env.DEFAULT_DB_VERSION == matrix.docker_db_version && env.SONAR_TOKEN != null }}
```

File: src/test/java/com/exasol/adapter/dialects/rls/AbstractRowLevelSecurityIT.java [hunk 6]
```diff
@@ -62,6 +64,11 @@ static void beforeAll() throws SQLException, BucketAccessException, InterruptedE
         createConnectionDefinition();
     }
 
+    @AfterAll
+    static void AfterAll() {
+
+    }
+
```
Comment (kaklakariada) [inline src/test/java/com/exasol/adapter/dialects/rls/AbstractRowLevelSecurityIT.java]:
```suggestion
```

File: src/test/java/com/exasol/adapter/dialects/rls/AbstractRowLevelSecurityIT.java [hunk 7]
```diff
@@ -88,19 +95,16 @@ private static void createConnectionDefinition() {
...
         return "jdbc:exa:localhost:" + port + ";validateservercertificate=0";
     }
 
-    protected static boolean exasolVersionSupportsFingerprintInAddress() {
-        final ExasolDockerImageReference imageReference = EXASOL.getDockerImageReference();
-        return (imageReference.getMajor() >= 7) && (imageReference.getMinor() >= 1);
-    }
 
     // [itest->dsn~query-rewriter-adds-row-filter-for-tenants~1]
+    @Tag("Specific")
```
Comment (kaklakariada) [inline src/test/java/com/exasol/adapter/dialects/rls/AbstractRowLevelSecurityIT.java]: Why did you add this?

File: src/test/java/com/exasol/adapter/dialects/rls/DBHelper.java [hunk 8]
```diff
@@ -0,0 +1,10 @@
+package com.exasol.adapter.dialects.rls;
+
+import com.exasol.containers.ExasolDockerImageReference;
+
+public class DBHelper {
+    public static boolean exasolVersionSupportsFingerprintInAddress(ExasolDockerImageReference exasolDockerImageReference) {
+        final ExasolDockerImageReference imageReference = exasolDockerImageReference;//
+        return (imageReference.getMajor() >= 7) && (imageReference.getMinor() >= 1);
+    }
```
Comment (kaklakariada) [inline src/test/java/com/exasol/adapter/dialects/rls/DBHelper.java]:
```suggestion
    public static boolean exasolVersionSupportsFingerprintInAddress(ExasolDockerImageReference imageReference) {
        return (imageReference.getMajor() >= 7) && (imageReference.getMinor() >= 1);
    }
```

File: src/test/java/com/exasol/adapter/dialects/rls/QueryRuntimeIT.java [hunk 9]
```diff
@@ -77,8 +78,15 @@ private static void createRowLevelSecuritySchema(final ExasolSchema sourceSchema
...
         final AdapterScript adapterScript = rlsSchema.createAdapterScript("RLS_VS_ADAPTER", JAVA, scriptContent);
-        final ConnectionDefinition connectionDefinition = objectFactory.createConnectionDefinition(
-                "EXASOL_JDBC_CONNECTION", "jdbc:exa:localhost:" + EXASOL.getExposedPorts().get(0), EXASOL.getUsername(),
+        final String connectionToDefinition;
+        if (exasolVersionSupportsFingerprintInAddress(EXASOL.getDockerImageReference())) {
+            final String fingerprint = EXASOL.getTlsCertificateFingerprint().get();
+            connectionToDefinition="jdbc:exa:localhost:" + EXASOL.getExposedPorts().get(0) + ";validateservercertificate=1;fingerprint="+fingerprint+";";
+        } else {
+            connectionToDefinition="jdbc:exa:localhost:" + EXASOL.getExposedPorts().get(0);
+        }
```
Comment (kaklakariada) [inline src/test/java/com/exasol/adapter/dialects/rls/QueryRuntimeIT.java]: Extract this into a method

File: src/assembly/all-dependencies.xml [hunk 10]
```diff
@@ -13,10 +13,23 @@
...
-            <scope>runtime</scope>
-            <outputDirectory>/</outputDirectory>
-        </dependencySet>
-    </dependencySets>
+    <dependencySet>
+        <outputDirectory>/</outputDirectory>
+        <useProjectArtifact>false</useProjectArtifact>
+        <unpack>true</unpack>
+        <unpackOptions>
+            <!-- We exclude this file here to change the order od registered dialects' factories: first should be RLS, then Exasol.-->
```
Comment (kaklakariada) [inline src/assembly/all-dependencies.xml]:
```suggestion
            <!-- We exclude this file here to change the order of registered dialects' factories: first should be RLS, then Exasol.-->
```

**Review by kaklakariada — CHANGES_REQUESTED** [review]

File: src/test/java/com/exasol/adapter/dialects/rls/AbstractRowLevelSecurityIT.java (hunk 7 above)
Comment (pj-spoelders) [reply src/test/java/com/exasol/adapter/dialects/rls/AbstractRowLevelSecurityIT.java]: Well  you weren't supposed to review this yet as it's WIP (Why are you reviewing this already? 😛 )
In this case we had a problem with container cleanup not working between some of the derived test classes.
We added the tag to have a faster test case. It's going to be removed soon :)

File: src/test/java/com/exasol/adapter/dialects/rls/AbstractRowLevelSecurityIT.java (hunk 6 above)
Comment (pj-spoelders) [reply src/test/java/com/exasol/adapter/dialects/rls/AbstractRowLevelSecurityIT.java]: Once again, WIP, it was going to get removed.

File: src/test/java/com/exasol/adapter/dialects/rls/DBHelper.java (hunk 8 above)
Comment (pj-spoelders) [reply src/test/java/com/exasol/adapter/dialects/rls/DBHelper.java]: Would've probably missed this, cleaned it up.

File: src/test/java/com/exasol/adapter/dialects/rls/QueryRuntimeIT.java (hunk 9 above)
Comment (pj-spoelders) [reply src/test/java/com/exasol/adapter/dialects/rls/QueryRuntimeIT.java]: Makes sense, done.

**Review by kaklakariada — APPROVED** [review]
